File: src/main/java/io/github/aveenstra/run_anything/RunAnythingConfigurationType.java
/*
 *    Copyright 2021 A Veenstra
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package io.github.aveenstra.run_anything;

import com.intellij.execution.configurations.ConfigurationFactory;
import com.intellij.execution.configurations.ConfigurationType;
import com.intellij.icons.AllIcons;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class RunAnythingConfigurationType implements ConfigurationType {

    @Override
    public @NotNull @Nls(capitalization = Nls.Capitalization.Title) String getDisplayName() {
        return "Run Command";
    }

    @Override
    public @Nls(capitalization = Nls.Capitalization.Sentence) String getConfigurationTypeDescription() {
        return "Run any system command.";
    }

    @Override
    public Icon getIcon() {
        return AllIcons.Actions.Run_anything;
    }

    @Override
    public @NotNull String getId() {
        return "RUN_ANYTHING_CONFIGURATION";
    }

    @Override
    public ConfigurationFactory[] getConfigurationFactories() {
        return new ConfigurationFactory[]{new RunAnythingConfigurationFactory(this)};
    }

    @Override
    public @Nullable String getHelpTopic() {
        return null;
    }
}

File: src/main/java/io/github/aveenstra/runAnything/RunAnythingConfigurationType.kt
/*
 *    Copyright 2022 A Veenstra
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */
package io.github.aveenstra.runAnything

import com.intellij.execution.configurations.ConfigurationType
import com.intellij.icons.AllIcons

/**
 * This class represents the configuration type of the plugin.
 */
class RunAnythingConfigurationType : ConfigurationType {
    override fun getDisplayName() = "Run Command"

    override fun getConfigurationTypeDescription() = "Run any system command."

    override fun getIcon() = AllIcons.Actions.Run_anything

    override fun getId() = "RUN_ANYTHING_CONFIGURATION"

    override fun getConfigurationFactories() = arrayOf(RunAnythingConfigurationFactory(this))

    override fun getHelpTopic(): Nothing? = null
}